File: src/foam/swift/parse/json/AnyParser.js
/**
 * @license
 * Copyright 2017 The FOAM Authors. All Rights Reserved.
 * http://www.apache.org/licenses/LICENSE-2.0
 */

foam.CLASS({
  package: 'foam.swift.parse.json',
  name: 'AnyParser',
  extends: 'foam.swift.parse.parser.ProxyParser',
  requires: [
    'foam.swift.parse.json.ArrayParser',
    'foam.swift.parse.json.BooleanParser',
    'foam.swift.parse.json.FObjectParser',
    'foam.swift.parse.json.FloatParser',
    'foam.swift.parse.json.IntParser',
    'foam.swift.parse.json.LongParser',
    'foam.swift.parse.json.MapParser',
    'foam.swift.parse.json.NullParser',
    'foam.swift.parse.json.StringParser',
    'foam.swift.parse.parser.Alt',
  ],
  properties: [
    {
      name: 'delegate',
      swiftFactory: function() {/*
return Alt_create(["parsers": [
  NullParser_create(),
  StringParser_create(),
  FloatParser_create(),
  LongParser_create(),
  IntParser_create(),
  BooleanParser_create(),
  FObjectParser_create(),
  ArrayParser_create(),
  MapParser_create(),
]])
      */},
    },
  ],
  axioms: [
    foam.pattern.Singleton.create()
  ],
});

File: src/foam/parse/AnyParser.java
/**
 * @license
 * Copyright 2018 The FOAM Authors. All Rights Reserved.
 * http://www.apache.org/licenses/LICENSE-2.0
 */

package foam.parse;

import foam.lib.json.BooleanParser;
import foam.lib.json.DoubleParser;
import foam.lib.json.LongParser;
import foam.lib.json.NullParser;
import foam.lib.parse.Alt;
import foam.lib.parse.ProxyParser;

public class AnyParser extends ProxyParser {
  private static AnyParser instance_ = null;

  public static AnyParser instance() {
    if (instance_ == null) {
      instance_ = new AnyParser();
      Alt delegate = new Alt(
          new NullParser(),
          new BooleanParser(),
          new LongParser(),
          new DoubleParser(),
          new FObjectParser(),
          new StringParser());
      instance_.setDelegate(delegate);
    }
    return instance_;
  }

  public AnyParser() {
  }
}
